controllers/ingressnodefirewall_controller.go

```diff
@@ -272,7 +275,7 @@ func (r *IngressNodeFirewallReconciler) buildNodeStates(
 		}
 
 	withNextNode:
-		for _, node := range nodeList.Items {
+		for _, node := range nodeList.DeepCopy().Items {
```

Comment:
you initialize nodeList in Line256 and populate it in Line272, I can't see another place where this can be mutated, do you need to deepcopy it?

Reply:
you are right not needed here